drivers/spi/spi_mcux_flexio.c

```diff
@@ -0,0 +1,446 @@
+	mcux_flexio_unlock(config->flexio_dev);
+
+	ret = spi_context_wait_for_completion(&data->ctx);
+out:
+	spi_context_release(&data->ctx, ret);
+
+	return ret;
+}
+
+static int spi_mcux_transceive(const struct device *dev,
+				const struct spi_config *spi_cfg,
+				const struct spi_buf_set *tx_bufs,
+				const struct spi_buf_set *rx_bufs)
+{
+	return transceive(dev, spi_cfg, tx_bufs, rx_bufs, false, NULL, NULL);
+}
+
+#ifdef CONFIG_SPI_ASYNC
+static int spi_mcux_transceive_async(const struct device *dev,
+				const struct spi_config *spi_cfg,
+				const struct spi_buf_set *tx_bufs,
+				const struct spi_buf_set *rx_bufs,
+				spi_callback_t cb,
+				void *userdata)
+{
+	return transceive(dev, spi_cfg, tx_bufs, rx_bufs, true, cb, userdata);
+}
+#endif /* CONFIG_SPI_ASYNC */
+
+static int spi_mcux_release(const struct device *dev,
+				const struct spi_config *spi_cfg)
+{
+	struct spi_mcux_flexio_data *data = dev->data;
+
+	spi_context_unlock_unconditionally(&data->ctx);
+
+	return 0;
+}
+
+static int spi_mcux_init(const struct device *dev)
+{
+	const struct spi_mcux_flexio_config *config = dev->config;
+	struct spi_mcux_flexio_data *data = dev->data;
+	int err;
+
+	err = spi_context_cs_configure_all(&data->ctx);
+	if (err < 0) {
+		return err;
+	}
+
+	spi_context_unlock_unconditionally(&data->ctx);
+
+	data->dev = dev;
+
+	/* TODO: DMA */
+
+#ifdef CONFIG_PINCTRL
+	err = pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
+	if (err) {
+		return err;
+	}
+#endif /* CONFIG_PINCTRL */
+
+	spi_context_unlock_unconditionally(&data->ctx);
+
+	return 0;
+}
+
+static const struct spi_driver_api spi_mcux_driver_api = {
+	.transceive = spi_mcux_transceive,
+#ifdef CONFIG_SPI_ASYNC
+	.transceive_async = spi_mcux_transceive_async,
+#endif
+	.release = spi_mcux_release,
+};
+
+#ifdef CONFIG_PINCTRL
+#define SPI_MCUX_FLEXIO_SPI_PINCTRL_DEFINE(n) PINCTRL_DT_INST_DEFINE(n);
+#define SPI_MCUX_FLEXIO_SPI_PINCTRL_INIT(n) .pincfg = PINCTRL_DT_INST_DEV_CONFIG_GET(n),
+#else
+#define SPI_MCUX_FLEXIO_SPI_PINCTRL_DEFINE(n)
+#define SPI_MCUX_FLEXIO_SPI_PINCTRL_INIT(n)
+#endif /* CONFIG_PINCTRL */
+
+
+#define SPI_MCUX_FLEXIO_SPI_INIT(n)					\
+	SPI_MCUX_FLEXIO_SPI_PINCTRL_DEFINE(n)				\
+									\
+	static FLEXIO_SPI_Type flexio_spi_##n = {			\
+		.flexioBase = (FLEXIO_Type *)DT_REG_ADDR(DT_INST_PARENT(n)), \
+		.SDOPinIndex = DT_INST_PROP(n, sdo_pin),		\
+		.SDIPinIndex = DT_INST_PROP(n, sdi_pin),		\
+		.SCKPinIndex = DT_INST_PROP(n, sck_pin),		\
+	};								\
+									\
+	static const struct spi_mcux_flexio_config spi_mcux_flexio_config_##n = { \
+		.flexio_spi = &flexio_spi_##n,				\
+		.flexio_dev = DEVICE_DT_GET(DT_INST_PARENT(n)),		\
+		SPI_MCUX_FLEXIO_SPI_PINCTRL_INIT(n)			\
+	};								\
+									\
+	static const STRUCT_SECTION_ITERABLE(mcux_flexio_child, mcux_flexio_child_##n) = { \
+		.dev = DEVICE_DT_INST_GET(n),				\
+		.isr = spi_mcux_flexio_isr,				\
+		.res = {						\
+			.shifter_index = flexio_spi_##n.shifterIndex,	\
```

Comment:
These were previously derived from Device Tree. How are you getting these now?

Reply:
@mmahadevan108, The FlexIO driver determines free resources and assigns them to the child: https://github.com/zephyrproject-rtos/zephyr/pull/53748/files#diff-1a74d72ae2e93b95139fdb5e22fbbac66251d0570a7f21c15571d7e191450fc6R160

Reply:
Sorry, I'm still a bit confused here- don't we want to take the address of these values here, IE `&flexsio_spi_##n.shifterIndex`?

Reply:
@danieldegrasse, That's right, `shifterIndex` is an array, and its name is a pointer to the zero element. This will be the same as `&flexsio_spi_##n.shifterIndex[0]`. https://github.com/zephyrproject-rtos/hal_nxp/blob/d45b14c198d778658b7853b48378d2e132a6c4be/mcux/mcux-sdk/drivers/flexio/spi/fsl_flexio_spi.h#L121

Reply:
Understood- sorry, I should have checked this closer. All good from my side then